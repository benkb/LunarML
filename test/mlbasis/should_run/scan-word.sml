functor TestWord (val name : string; structure W : WORD) = struct
fun printOptionWord NONE = print "NONE\n"
  | printOptionWord (SOME w) = print ("SOME 0w" ^ W.fmt StringCvt.DEC w ^ "\n")
val ss = ["", "0wx0", "0wx1", "0wX10", "99", "100", "0w", "0wx", "0xa", "0Xg", "0W3", "0", "10", "100", "1020", "1080", "0wx2", "8", "fF", " \t11"];
val () = List.app (fn s => (print (name ^ ".fromString \"" ^ String.toString s ^ "\" = "); printOptionWord (W.fromString s) handle Overflow => print "Overflow\n")) ss
val () = List.app (fn (radixs, radix) => List.app (fn s => (print (name ^ ".scan " ^ radixs ^ " \"" ^ String.toString s ^ "\" = "); printOptionWord (StringCvt.scanString (W.scan radix) s) handle Overflow => print "Overflow\n")) ss) [("BIN", StringCvt.BIN), ("OCT", StringCvt.OCT), ("DEC", StringCvt.DEC), ("HEX", StringCvt.HEX)]
end;
structure TestWord = TestWord (val name = "Word"; structure W = Word)
structure TestWord8 = TestWord (val name = "Word8"; structure W = Word8)
structure TestWord16 = TestWord (val name = "Word16"; structure W = Word16)
structure TestWord32 = TestWord (val name = "Word32"; structure W = Word32)
structure TestWord64 = TestWord (val name = "Word64"; structure W = Word64);
