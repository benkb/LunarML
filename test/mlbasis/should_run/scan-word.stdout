Word.fromString "" = NONE
Word.fromString "0wx0" = SOME 0w0
Word.fromString "0wx1" = SOME 0w1
Word.fromString "0wX10" = SOME 0w16
Word.fromString "99" = SOME 0w153
Word.fromString "100" = SOME 0w256
Word.fromString "0w" = SOME 0w0
Word.fromString "0wx" = SOME 0w0
Word.fromString "0xa" = SOME 0w10
Word.fromString "0Xg" = SOME 0w0
Word.fromString "0W3" = SOME 0w0
Word.fromString "0" = SOME 0w0
Word.fromString "10" = SOME 0w16
Word.fromString "100" = SOME 0w256
Word.fromString "1020" = SOME 0w4128
Word.fromString "1080" = SOME 0w4224
Word.fromString "0wx2" = SOME 0w2
Word.fromString "8" = SOME 0w8
Word.fromString "fF" = SOME 0w255
Word.fromString " \t11" = SOME 0w17
Word.scan BIN "" = NONE
Word.scan BIN "0wx0" = SOME 0w0
Word.scan BIN "0wx1" = SOME 0w0
Word.scan BIN "0wX10" = SOME 0w0
Word.scan BIN "99" = NONE
Word.scan BIN "100" = SOME 0w4
Word.scan BIN "0w" = SOME 0w0
Word.scan BIN "0wx" = SOME 0w0
Word.scan BIN "0xa" = SOME 0w0
Word.scan BIN "0Xg" = SOME 0w0
Word.scan BIN "0W3" = SOME 0w0
Word.scan BIN "0" = SOME 0w0
Word.scan BIN "10" = SOME 0w2
Word.scan BIN "100" = SOME 0w4
Word.scan BIN "1020" = SOME 0w2
Word.scan BIN "1080" = SOME 0w2
Word.scan BIN "0wx2" = SOME 0w0
Word.scan BIN "8" = NONE
Word.scan BIN "fF" = NONE
Word.scan BIN " \t11" = SOME 0w3
Word.scan OCT "" = NONE
Word.scan OCT "0wx0" = SOME 0w0
Word.scan OCT "0wx1" = SOME 0w0
Word.scan OCT "0wX10" = SOME 0w0
Word.scan OCT "99" = NONE
Word.scan OCT "100" = SOME 0w64
Word.scan OCT "0w" = SOME 0w0
Word.scan OCT "0wx" = SOME 0w0
Word.scan OCT "0xa" = SOME 0w0
Word.scan OCT "0Xg" = SOME 0w0
Word.scan OCT "0W3" = SOME 0w0
Word.scan OCT "0" = SOME 0w0
Word.scan OCT "10" = SOME 0w8
Word.scan OCT "100" = SOME 0w64
Word.scan OCT "1020" = SOME 0w528
Word.scan OCT "1080" = SOME 0w8
Word.scan OCT "0wx2" = SOME 0w0
Word.scan OCT "8" = NONE
Word.scan OCT "fF" = NONE
Word.scan OCT " \t11" = SOME 0w9
Word.scan DEC "" = NONE
Word.scan DEC "0wx0" = SOME 0w0
Word.scan DEC "0wx1" = SOME 0w0
Word.scan DEC "0wX10" = SOME 0w0
Word.scan DEC "99" = SOME 0w99
Word.scan DEC "100" = SOME 0w100
Word.scan DEC "0w" = SOME 0w0
Word.scan DEC "0wx" = SOME 0w0
Word.scan DEC "0xa" = SOME 0w0
Word.scan DEC "0Xg" = SOME 0w0
Word.scan DEC "0W3" = SOME 0w0
Word.scan DEC "0" = SOME 0w0
Word.scan DEC "10" = SOME 0w10
Word.scan DEC "100" = SOME 0w100
Word.scan DEC "1020" = SOME 0w1020
Word.scan DEC "1080" = SOME 0w1080
Word.scan DEC "0wx2" = SOME 0w0
Word.scan DEC "8" = SOME 0w8
Word.scan DEC "fF" = NONE
Word.scan DEC " \t11" = SOME 0w11
Word.scan HEX "" = NONE
Word.scan HEX "0wx0" = SOME 0w0
Word.scan HEX "0wx1" = SOME 0w1
Word.scan HEX "0wX10" = SOME 0w16
Word.scan HEX "99" = SOME 0w153
Word.scan HEX "100" = SOME 0w256
Word.scan HEX "0w" = SOME 0w0
Word.scan HEX "0wx" = SOME 0w0
Word.scan HEX "0xa" = SOME 0w10
Word.scan HEX "0Xg" = SOME 0w0
Word.scan HEX "0W3" = SOME 0w0
Word.scan HEX "0" = SOME 0w0
Word.scan HEX "10" = SOME 0w16
Word.scan HEX "100" = SOME 0w256
Word.scan HEX "1020" = SOME 0w4128
Word.scan HEX "1080" = SOME 0w4224
Word.scan HEX "0wx2" = SOME 0w2
Word.scan HEX "8" = SOME 0w8
Word.scan HEX "fF" = SOME 0w255
Word.scan HEX " \t11" = SOME 0w17
Word8.fromString "" = NONE
Word8.fromString "0wx0" = SOME 0w0
Word8.fromString "0wx1" = SOME 0w1
Word8.fromString "0wX10" = SOME 0w16
Word8.fromString "99" = SOME 0w153
Word8.fromString "100" = Overflow
Word8.fromString "0w" = SOME 0w0
Word8.fromString "0wx" = SOME 0w0
Word8.fromString "0xa" = SOME 0w10
Word8.fromString "0Xg" = SOME 0w0
Word8.fromString "0W3" = SOME 0w0
Word8.fromString "0" = SOME 0w0
Word8.fromString "10" = SOME 0w16
Word8.fromString "100" = Overflow
Word8.fromString "1020" = Overflow
Word8.fromString "1080" = Overflow
Word8.fromString "0wx2" = SOME 0w2
Word8.fromString "8" = SOME 0w8
Word8.fromString "fF" = SOME 0w255
Word8.fromString " \t11" = SOME 0w17
Word8.scan BIN "" = NONE
Word8.scan BIN "0wx0" = SOME 0w0
Word8.scan BIN "0wx1" = SOME 0w0
Word8.scan BIN "0wX10" = SOME 0w0
Word8.scan BIN "99" = NONE
Word8.scan BIN "100" = SOME 0w4
Word8.scan BIN "0w" = SOME 0w0
Word8.scan BIN "0wx" = SOME 0w0
Word8.scan BIN "0xa" = SOME 0w0
Word8.scan BIN "0Xg" = SOME 0w0
Word8.scan BIN "0W3" = SOME 0w0
Word8.scan BIN "0" = SOME 0w0
Word8.scan BIN "10" = SOME 0w2
Word8.scan BIN "100" = SOME 0w4
Word8.scan BIN "1020" = SOME 0w2
Word8.scan BIN "1080" = SOME 0w2
Word8.scan BIN "0wx2" = SOME 0w0
Word8.scan BIN "8" = NONE
Word8.scan BIN "fF" = NONE
Word8.scan BIN " \t11" = SOME 0w3
Word8.scan OCT "" = NONE
Word8.scan OCT "0wx0" = SOME 0w0
Word8.scan OCT "0wx1" = SOME 0w0
Word8.scan OCT "0wX10" = SOME 0w0
Word8.scan OCT "99" = NONE
Word8.scan OCT "100" = SOME 0w64
Word8.scan OCT "0w" = SOME 0w0
Word8.scan OCT "0wx" = SOME 0w0
Word8.scan OCT "0xa" = SOME 0w0
Word8.scan OCT "0Xg" = SOME 0w0
Word8.scan OCT "0W3" = SOME 0w0
Word8.scan OCT "0" = SOME 0w0
Word8.scan OCT "10" = SOME 0w8
Word8.scan OCT "100" = SOME 0w64
Word8.scan OCT "1020" = Overflow
Word8.scan OCT "1080" = SOME 0w8
Word8.scan OCT "0wx2" = SOME 0w0
Word8.scan OCT "8" = NONE
Word8.scan OCT "fF" = NONE
Word8.scan OCT " \t11" = SOME 0w9
Word8.scan DEC "" = NONE
Word8.scan DEC "0wx0" = SOME 0w0
Word8.scan DEC "0wx1" = SOME 0w0
Word8.scan DEC "0wX10" = SOME 0w0
Word8.scan DEC "99" = SOME 0w99
Word8.scan DEC "100" = SOME 0w100
Word8.scan DEC "0w" = SOME 0w0
Word8.scan DEC "0wx" = SOME 0w0
Word8.scan DEC "0xa" = SOME 0w0
Word8.scan DEC "0Xg" = SOME 0w0
Word8.scan DEC "0W3" = SOME 0w0
Word8.scan DEC "0" = SOME 0w0
Word8.scan DEC "10" = SOME 0w10
Word8.scan DEC "100" = SOME 0w100
Word8.scan DEC "1020" = Overflow
Word8.scan DEC "1080" = Overflow
Word8.scan DEC "0wx2" = SOME 0w0
Word8.scan DEC "8" = SOME 0w8
Word8.scan DEC "fF" = NONE
Word8.scan DEC " \t11" = SOME 0w11
Word8.scan HEX "" = NONE
Word8.scan HEX "0wx0" = SOME 0w0
Word8.scan HEX "0wx1" = SOME 0w1
Word8.scan HEX "0wX10" = SOME 0w16
Word8.scan HEX "99" = SOME 0w153
Word8.scan HEX "100" = Overflow
Word8.scan HEX "0w" = SOME 0w0
Word8.scan HEX "0wx" = SOME 0w0
Word8.scan HEX "0xa" = SOME 0w10
Word8.scan HEX "0Xg" = SOME 0w0
Word8.scan HEX "0W3" = SOME 0w0
Word8.scan HEX "0" = SOME 0w0
Word8.scan HEX "10" = SOME 0w16
Word8.scan HEX "100" = Overflow
Word8.scan HEX "1020" = Overflow
Word8.scan HEX "1080" = Overflow
Word8.scan HEX "0wx2" = SOME 0w2
Word8.scan HEX "8" = SOME 0w8
Word8.scan HEX "fF" = SOME 0w255
Word8.scan HEX " \t11" = SOME 0w17
Word16.fromString "" = NONE
Word16.fromString "0wx0" = SOME 0w0
Word16.fromString "0wx1" = SOME 0w1
Word16.fromString "0wX10" = SOME 0w16
Word16.fromString "99" = SOME 0w153
Word16.fromString "100" = SOME 0w256
Word16.fromString "0w" = SOME 0w0
Word16.fromString "0wx" = SOME 0w0
Word16.fromString "0xa" = SOME 0w10
Word16.fromString "0Xg" = SOME 0w0
Word16.fromString "0W3" = SOME 0w0
Word16.fromString "0" = SOME 0w0
Word16.fromString "10" = SOME 0w16
Word16.fromString "100" = SOME 0w256
Word16.fromString "1020" = SOME 0w4128
Word16.fromString "1080" = SOME 0w4224
Word16.fromString "0wx2" = SOME 0w2
Word16.fromString "8" = SOME 0w8
Word16.fromString "fF" = SOME 0w255
Word16.fromString " \t11" = SOME 0w17
Word16.scan BIN "" = NONE
Word16.scan BIN "0wx0" = SOME 0w0
Word16.scan BIN "0wx1" = SOME 0w0
Word16.scan BIN "0wX10" = SOME 0w0
Word16.scan BIN "99" = NONE
Word16.scan BIN "100" = SOME 0w4
Word16.scan BIN "0w" = SOME 0w0
Word16.scan BIN "0wx" = SOME 0w0
Word16.scan BIN "0xa" = SOME 0w0
Word16.scan BIN "0Xg" = SOME 0w0
Word16.scan BIN "0W3" = SOME 0w0
Word16.scan BIN "0" = SOME 0w0
Word16.scan BIN "10" = SOME 0w2
Word16.scan BIN "100" = SOME 0w4
Word16.scan BIN "1020" = SOME 0w2
Word16.scan BIN "1080" = SOME 0w2
Word16.scan BIN "0wx2" = SOME 0w0
Word16.scan BIN "8" = NONE
Word16.scan BIN "fF" = NONE
Word16.scan BIN " \t11" = SOME 0w3
Word16.scan OCT "" = NONE
Word16.scan OCT "0wx0" = SOME 0w0
Word16.scan OCT "0wx1" = SOME 0w0
Word16.scan OCT "0wX10" = SOME 0w0
Word16.scan OCT "99" = NONE
Word16.scan OCT "100" = SOME 0w64
Word16.scan OCT "0w" = SOME 0w0
Word16.scan OCT "0wx" = SOME 0w0
Word16.scan OCT "0xa" = SOME 0w0
Word16.scan OCT "0Xg" = SOME 0w0
Word16.scan OCT "0W3" = SOME 0w0
Word16.scan OCT "0" = SOME 0w0
Word16.scan OCT "10" = SOME 0w8
Word16.scan OCT "100" = SOME 0w64
Word16.scan OCT "1020" = SOME 0w528
Word16.scan OCT "1080" = SOME 0w8
Word16.scan OCT "0wx2" = SOME 0w0
Word16.scan OCT "8" = NONE
Word16.scan OCT "fF" = NONE
Word16.scan OCT " \t11" = SOME 0w9
Word16.scan DEC "" = NONE
Word16.scan DEC "0wx0" = SOME 0w0
Word16.scan DEC "0wx1" = SOME 0w0
Word16.scan DEC "0wX10" = SOME 0w0
Word16.scan DEC "99" = SOME 0w99
Word16.scan DEC "100" = SOME 0w100
Word16.scan DEC "0w" = SOME 0w0
Word16.scan DEC "0wx" = SOME 0w0
Word16.scan DEC "0xa" = SOME 0w0
Word16.scan DEC "0Xg" = SOME 0w0
Word16.scan DEC "0W3" = SOME 0w0
Word16.scan DEC "0" = SOME 0w0
Word16.scan DEC "10" = SOME 0w10
Word16.scan DEC "100" = SOME 0w100
Word16.scan DEC "1020" = SOME 0w1020
Word16.scan DEC "1080" = SOME 0w1080
Word16.scan DEC "0wx2" = SOME 0w0
Word16.scan DEC "8" = SOME 0w8
Word16.scan DEC "fF" = NONE
Word16.scan DEC " \t11" = SOME 0w11
Word16.scan HEX "" = NONE
Word16.scan HEX "0wx0" = SOME 0w0
Word16.scan HEX "0wx1" = SOME 0w1
Word16.scan HEX "0wX10" = SOME 0w16
Word16.scan HEX "99" = SOME 0w153
Word16.scan HEX "100" = SOME 0w256
Word16.scan HEX "0w" = SOME 0w0
Word16.scan HEX "0wx" = SOME 0w0
Word16.scan HEX "0xa" = SOME 0w10
Word16.scan HEX "0Xg" = SOME 0w0
Word16.scan HEX "0W3" = SOME 0w0
Word16.scan HEX "0" = SOME 0w0
Word16.scan HEX "10" = SOME 0w16
Word16.scan HEX "100" = SOME 0w256
Word16.scan HEX "1020" = SOME 0w4128
Word16.scan HEX "1080" = SOME 0w4224
Word16.scan HEX "0wx2" = SOME 0w2
Word16.scan HEX "8" = SOME 0w8
Word16.scan HEX "fF" = SOME 0w255
Word16.scan HEX " \t11" = SOME 0w17
Word32.fromString "" = NONE
Word32.fromString "0wx0" = SOME 0w0
Word32.fromString "0wx1" = SOME 0w1
Word32.fromString "0wX10" = SOME 0w16
Word32.fromString "99" = SOME 0w153
Word32.fromString "100" = SOME 0w256
Word32.fromString "0w" = SOME 0w0
Word32.fromString "0wx" = SOME 0w0
Word32.fromString "0xa" = SOME 0w10
Word32.fromString "0Xg" = SOME 0w0
Word32.fromString "0W3" = SOME 0w0
Word32.fromString "0" = SOME 0w0
Word32.fromString "10" = SOME 0w16
Word32.fromString "100" = SOME 0w256
Word32.fromString "1020" = SOME 0w4128
Word32.fromString "1080" = SOME 0w4224
Word32.fromString "0wx2" = SOME 0w2
Word32.fromString "8" = SOME 0w8
Word32.fromString "fF" = SOME 0w255
Word32.fromString " \t11" = SOME 0w17
Word32.scan BIN "" = NONE
Word32.scan BIN "0wx0" = SOME 0w0
Word32.scan BIN "0wx1" = SOME 0w0
Word32.scan BIN "0wX10" = SOME 0w0
Word32.scan BIN "99" = NONE
Word32.scan BIN "100" = SOME 0w4
Word32.scan BIN "0w" = SOME 0w0
Word32.scan BIN "0wx" = SOME 0w0
Word32.scan BIN "0xa" = SOME 0w0
Word32.scan BIN "0Xg" = SOME 0w0
Word32.scan BIN "0W3" = SOME 0w0
Word32.scan BIN "0" = SOME 0w0
Word32.scan BIN "10" = SOME 0w2
Word32.scan BIN "100" = SOME 0w4
Word32.scan BIN "1020" = SOME 0w2
Word32.scan BIN "1080" = SOME 0w2
Word32.scan BIN "0wx2" = SOME 0w0
Word32.scan BIN "8" = NONE
Word32.scan BIN "fF" = NONE
Word32.scan BIN " \t11" = SOME 0w3
Word32.scan OCT "" = NONE
Word32.scan OCT "0wx0" = SOME 0w0
Word32.scan OCT "0wx1" = SOME 0w0
Word32.scan OCT "0wX10" = SOME 0w0
Word32.scan OCT "99" = NONE
Word32.scan OCT "100" = SOME 0w64
Word32.scan OCT "0w" = SOME 0w0
Word32.scan OCT "0wx" = SOME 0w0
Word32.scan OCT "0xa" = SOME 0w0
Word32.scan OCT "0Xg" = SOME 0w0
Word32.scan OCT "0W3" = SOME 0w0
Word32.scan OCT "0" = SOME 0w0
Word32.scan OCT "10" = SOME 0w8
Word32.scan OCT "100" = SOME 0w64
Word32.scan OCT "1020" = SOME 0w528
Word32.scan OCT "1080" = SOME 0w8
Word32.scan OCT "0wx2" = SOME 0w0
Word32.scan OCT "8" = NONE
Word32.scan OCT "fF" = NONE
Word32.scan OCT " \t11" = SOME 0w9
Word32.scan DEC "" = NONE
Word32.scan DEC "0wx0" = SOME 0w0
Word32.scan DEC "0wx1" = SOME 0w0
Word32.scan DEC "0wX10" = SOME 0w0
Word32.scan DEC "99" = SOME 0w99
Word32.scan DEC "100" = SOME 0w100
Word32.scan DEC "0w" = SOME 0w0
Word32.scan DEC "0wx" = SOME 0w0
Word32.scan DEC "0xa" = SOME 0w0
Word32.scan DEC "0Xg" = SOME 0w0
Word32.scan DEC "0W3" = SOME 0w0
Word32.scan DEC "0" = SOME 0w0
Word32.scan DEC "10" = SOME 0w10
Word32.scan DEC "100" = SOME 0w100
Word32.scan DEC "1020" = SOME 0w1020
Word32.scan DEC "1080" = SOME 0w1080
Word32.scan DEC "0wx2" = SOME 0w0
Word32.scan DEC "8" = SOME 0w8
Word32.scan DEC "fF" = NONE
Word32.scan DEC " \t11" = SOME 0w11
Word32.scan HEX "" = NONE
Word32.scan HEX "0wx0" = SOME 0w0
Word32.scan HEX "0wx1" = SOME 0w1
Word32.scan HEX "0wX10" = SOME 0w16
Word32.scan HEX "99" = SOME 0w153
Word32.scan HEX "100" = SOME 0w256
Word32.scan HEX "0w" = SOME 0w0
Word32.scan HEX "0wx" = SOME 0w0
Word32.scan HEX "0xa" = SOME 0w10
Word32.scan HEX "0Xg" = SOME 0w0
Word32.scan HEX "0W3" = SOME 0w0
Word32.scan HEX "0" = SOME 0w0
Word32.scan HEX "10" = SOME 0w16
Word32.scan HEX "100" = SOME 0w256
Word32.scan HEX "1020" = SOME 0w4128
Word32.scan HEX "1080" = SOME 0w4224
Word32.scan HEX "0wx2" = SOME 0w2
Word32.scan HEX "8" = SOME 0w8
Word32.scan HEX "fF" = SOME 0w255
Word32.scan HEX " \t11" = SOME 0w17
Word64.fromString "" = NONE
Word64.fromString "0wx0" = SOME 0w0
Word64.fromString "0wx1" = SOME 0w1
Word64.fromString "0wX10" = SOME 0w16
Word64.fromString "99" = SOME 0w153
Word64.fromString "100" = SOME 0w256
Word64.fromString "0w" = SOME 0w0
Word64.fromString "0wx" = SOME 0w0
Word64.fromString "0xa" = SOME 0w10
Word64.fromString "0Xg" = SOME 0w0
Word64.fromString "0W3" = SOME 0w0
Word64.fromString "0" = SOME 0w0
Word64.fromString "10" = SOME 0w16
Word64.fromString "100" = SOME 0w256
Word64.fromString "1020" = SOME 0w4128
Word64.fromString "1080" = SOME 0w4224
Word64.fromString "0wx2" = SOME 0w2
Word64.fromString "8" = SOME 0w8
Word64.fromString "fF" = SOME 0w255
Word64.fromString " \t11" = SOME 0w17
Word64.scan BIN "" = NONE
Word64.scan BIN "0wx0" = SOME 0w0
Word64.scan BIN "0wx1" = SOME 0w0
Word64.scan BIN "0wX10" = SOME 0w0
Word64.scan BIN "99" = NONE
Word64.scan BIN "100" = SOME 0w4
Word64.scan BIN "0w" = SOME 0w0
Word64.scan BIN "0wx" = SOME 0w0
Word64.scan BIN "0xa" = SOME 0w0
Word64.scan BIN "0Xg" = SOME 0w0
Word64.scan BIN "0W3" = SOME 0w0
Word64.scan BIN "0" = SOME 0w0
Word64.scan BIN "10" = SOME 0w2
Word64.scan BIN "100" = SOME 0w4
Word64.scan BIN "1020" = SOME 0w2
Word64.scan BIN "1080" = SOME 0w2
Word64.scan BIN "0wx2" = SOME 0w0
Word64.scan BIN "8" = NONE
Word64.scan BIN "fF" = NONE
Word64.scan BIN " \t11" = SOME 0w3
Word64.scan OCT "" = NONE
Word64.scan OCT "0wx0" = SOME 0w0
Word64.scan OCT "0wx1" = SOME 0w0
Word64.scan OCT "0wX10" = SOME 0w0
Word64.scan OCT "99" = NONE
Word64.scan OCT "100" = SOME 0w64
Word64.scan OCT "0w" = SOME 0w0
Word64.scan OCT "0wx" = SOME 0w0
Word64.scan OCT "0xa" = SOME 0w0
Word64.scan OCT "0Xg" = SOME 0w0
Word64.scan OCT "0W3" = SOME 0w0
Word64.scan OCT "0" = SOME 0w0
Word64.scan OCT "10" = SOME 0w8
Word64.scan OCT "100" = SOME 0w64
Word64.scan OCT "1020" = SOME 0w528
Word64.scan OCT "1080" = SOME 0w8
Word64.scan OCT "0wx2" = SOME 0w0
Word64.scan OCT "8" = NONE
Word64.scan OCT "fF" = NONE
Word64.scan OCT " \t11" = SOME 0w9
Word64.scan DEC "" = NONE
Word64.scan DEC "0wx0" = SOME 0w0
Word64.scan DEC "0wx1" = SOME 0w0
Word64.scan DEC "0wX10" = SOME 0w0
Word64.scan DEC "99" = SOME 0w99
Word64.scan DEC "100" = SOME 0w100
Word64.scan DEC "0w" = SOME 0w0
Word64.scan DEC "0wx" = SOME 0w0
Word64.scan DEC "0xa" = SOME 0w0
Word64.scan DEC "0Xg" = SOME 0w0
Word64.scan DEC "0W3" = SOME 0w0
Word64.scan DEC "0" = SOME 0w0
Word64.scan DEC "10" = SOME 0w10
Word64.scan DEC "100" = SOME 0w100
Word64.scan DEC "1020" = SOME 0w1020
Word64.scan DEC "1080" = SOME 0w1080
Word64.scan DEC "0wx2" = SOME 0w0
Word64.scan DEC "8" = SOME 0w8
Word64.scan DEC "fF" = NONE
Word64.scan DEC " \t11" = SOME 0w11
Word64.scan HEX "" = NONE
Word64.scan HEX "0wx0" = SOME 0w0
Word64.scan HEX "0wx1" = SOME 0w1
Word64.scan HEX "0wX10" = SOME 0w16
Word64.scan HEX "99" = SOME 0w153
Word64.scan HEX "100" = SOME 0w256
Word64.scan HEX "0w" = SOME 0w0
Word64.scan HEX "0wx" = SOME 0w0
Word64.scan HEX "0xa" = SOME 0w10
Word64.scan HEX "0Xg" = SOME 0w0
Word64.scan HEX "0W3" = SOME 0w0
Word64.scan HEX "0" = SOME 0w0
Word64.scan HEX "10" = SOME 0w16
Word64.scan HEX "100" = SOME 0w256
Word64.scan HEX "1020" = SOME 0w4128
Word64.scan HEX "1080" = SOME 0w4224
Word64.scan HEX "0wx2" = SOME 0w2
Word64.scan HEX "8" = SOME 0w8
Word64.scan HEX "fF" = SOME 0w255
Word64.scan HEX " \t11" = SOME 0w17
